scss/mixins/_grid-framework.scss

```diff
@@ -1,58 +1,65 @@
-// Grid Framework Generation
-//––––––––––––––––––––––––––––––––––––––––––––––––––
-
-// Used to generate the correct number of grid classes given any value of `$grid-columns`.
+// Framework grid generation
+//
+// Used only by Bootstrap to generate the correct number of grid classes given
+// any value of `$grid-columns`.
 
 @mixin make-grid-columns($columns: $grid-columns, $gutters: $grid-gutter-widths, $breakpoints: $grid-breakpoints) {
-    
-    // Common properties for all breakpoints
-    %grid-column {
-        position: relative;
-        width: 100%;
-        min-height: 1px;
-
-        @include make-gutters($gutters);
+  // Common properties for all breakpoints
+  %grid-column {
+    position: relative;
+    width: 100%;
+    min-height: 1px; // Prevent columns from collapsing when empty
+
+    @include make-gutters($gutters);
+  }
+
+  @each $breakpoint in map-keys($breakpoints) {
+    $infix: breakpoint-infix($breakpoint, $breakpoints);
+
+    // Allow columns to stretch full width below their breakpoints
+    @for $i from 1 through $columns {
+      .col#{$infix}-#{$i} {
+        @extend %grid-column;
+      }
+    }
+    .col#{$infix} {
+      @extend %grid-column;
     }
 
-    @each $breakpoint in map-keys($breakpoints) {
-        $infix: breakpoint-infix($breakpoint, $breakpoints);
+    @include media-breakpoint-up($breakpoint, $breakpoints) {
+      // Provide basic `.col-{bp}` classes for equal-width flexbox columns
+      .col#{$infix} {
+        flex-basis: 0;
+        flex-grow: 1;
+        max-width: 100%;
+      }
+      .col#{$infix}-auto {
+        flex: 0 0 auto;
+        width: auto;
+      }
 
-        // Allow columns to stretch full width below their breakpoints
-        @for $i from 1 through $columns {
-            .col#{$infix}-#{$i} {
-                @extend %grid-column;
-            }
+      @for $i from 1 through $columns {
+        .col#{$infix}-#{$i} {
+          @include make-col($i, $columns);
         }
-        .col#{$infix},
-        .col#{$infix}-auto {
-            @extend %grid-column;
+      }
+
+      @each $modifier in (pull, push) {
```

Comment:
you have re-introduced a bug here that outputted invalid styles see other comment